src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -323,6 +323,12 @@ function VisEditor($rootScope, $scope, $route, timefilter, AppState, $window, kb
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
         if ($scope.isAddToDashMode()) {
+          // We need to manually set the last url for this app so navigating back to visualize will open the saved
+          // visualization, not the unsaved one.
+          const savedVisualizationUrl =
+            kbnUrl.eval(`${kbnBaseUrl}#${VisualizeConstants.EDIT_PATH}/{{id}}`, { id: savedVis.id });
+          chrome.trackSubUrlForApp('kibana:visualize', savedVisualizationUrl);
```

Comment:
`chrome. trackPossibleSubUrl()` gets called [when there is a route update](https://github.com/elastic/kibana/blob/eb31af6d7b7c91be26b914f3cf998bb0e89e159d/src/ui/public/chrome/directives/kbn_chrome.js#L42-L51) which should always occur when using `kbnUrl.redirect(newUrl)` but it takes a digest cycle before it happens.